scripts/ci/check-tutorials-index.py

```diff
@@ -0,0 +1,107 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Check all notebooks are listed on the tutorials index page, and that their
+titles match the titles in the notebook.
+"""
+
+from pathlib import Path
+from dataclasses import dataclass
+import json
+import re
+
+TUTORIALS_ROOT = Path("docs/tutorials")
+INDEX_PAGE = Path("docs/tutorials/index.mdx")
+
+@dataclass
+class Link:
+    link_text: str
+    href: str
+
+    def __str__(self) -> str:
+        return f"[{self.link_text}]({self.href})"
+
+def is_hidden(path: Path) -> bool:
+    return any(part.startswith(".") for part in path.parts)
+
+def get_notebook_title(path: Path) -> str:
+    data = json.loads(path.read_text())
+    return data['metadata']['title'].strip()
+
+def get_expected_links() -> list[Link]:
+    notebook_paths = TUTORIALS_ROOT.rglob("**/*.ipynb")
+    return [
+        Link(
+            href=f"/{path.with_suffix('')}",
+            link_text=get_notebook_title(path)
+        )
+        for path in notebook_paths
+        if not is_hidden(path)
+    ]
+
+def extract_markdown_links(md: str) -> Link:
```

Comment:
Return type is wrong. This is returning a list[Link]` (and could be `Iterator[Link]` if you use a generator expression)